tests/helpers/flows/nc.ts

```diff
@@ -0,0 +1,198 @@
+    const primaryUserResult = await completePrimaryUserInfo(page, data.primaryUser);
+    if (!primaryUserResult.success) return primaryUserResult;
+    
+    // Complete household member info
+    const memberResult = await completeHouseholdMemberInfo(page, data.householdMember);
+    if (!memberResult.success) return memberResult;
+    
+    // Complete expenses
+    const expensesResult = await completeExpenses(page, data.expenses);
+    if (!expensesResult.success) return expensesResult;
+    
+    // Complete assets
+    const assetsResult = await completeAssets(page, data.assets);
+    if (!assetsResult.success) return assetsResult;
+    
+    // Complete public benefits (just continue)
+    const benefitsResult = await completePublicBenefits(page);
+    if (!benefitsResult.success) return benefitsResult;
+    
+    // Complete needs
+    const needsResult = await completeNeeds(page, data.needs);
+    if (!needsResult.success) return needsResult;
+    
+    // Complete referral source
+    const referralResult = await completeReferralSource(page, data.referralSource);
+    if (!referralResult.success) return referralResult;
+    
+    // Complete additional info
+    const additionalInfoResult = await completeAdditionalInfo(page);
+    if (!additionalInfoResult.success) return additionalInfoResult;
+    
+    // Navigate to results
+    const resultsResult = await navigateToResults(page);
+    if (!resultsResult.success) return resultsResult;
+    
+    return { success: true, step: 'nc-full-application' };
+  } catch (error) {
+    return {
+      success: false,
+      step: 'nc-full-application',
+      error: error as Error
+    };
+  }
+}
+
+/**
+ * Runs a complete end-to-end test for the NC white label
+ * @param page - Playwright page instance
+ * @param data - Application data to use for the test
+ * @returns Promise with flow result
+ */
+export async function runNcEndToEndTest(
+  page: Page,
+  data: ApplicationData
+): Promise<FlowResult> {
+  // Add retry mechanism for more resilience
+  const maxRetries = 2;
+  let attempt = 0;
+  let lastError: Error | null = null;
+
+  // Configure longer timeout for page operations
+  page.setDefaultTimeout(60000); // 60 seconds timeout for all operations
+
+  while (attempt <= maxRetries) {
+    try {
+      if (attempt > 0) {
+        console.log(`Retry attempt ${attempt} for NC end-to-end test`);
+        // Navigate back to home page for a fresh start on retry
+        await page.goto('/');
+        await page.waitForLoadState('networkidle');
+      }
+
+      // Setup session (landing page, state selection, disclaimer)
+      const setupResult = await setupNcSession(page);
+      if (!setupResult.success) {
+        lastError = setupResult.error || new Error('Unknown error in setup');
+        attempt++;
+        continue;
+      }
+      
+      // Complete full application with additional waiting for stability
+      await page.waitForLoadState('networkidle');
```

Comment:
In the docs I see useing networkidle is discouraged, any thoughts if it is worth it here?